### PR TITLE
Ubuntu14 upgrade fix

### PR DESCRIPTION
**Pull request by amoskong**
Tested by http://jenkins.cloudius-systems.com:8080/job/scylla-3.0/job/3.0-test/job/scylla-rolling-upgrade-ubuntu-14.04-test/

### CODE REVIEW

**Comment by amoskong** [comment]
Tested this PR by:
Ubuntu 14.04 (enable sst3): (PASS)
http://jenkins.cloudius-systems.com:8080/job/scylla-3.0/job/3.0-test/job/scylla-rolling-upgrade-ubuntu-14.04-test/11/
Ubuntu 14.04: (PASS)
http://jenkins.cloudius-systems.com:8080/job/scylla-3.0/job/3.0-test/job/scylla-rolling-upgrade-ubuntu-14.04-test/12/

File: sdcm/collectd.py [hunk 1]
```diff
@@ -404,7 +404,7 @@ def start_collectd_service(self):
             if self.node.is_rhel_like() or self.node.is_ubuntu16():
                 self.node.remoter.run('sudo systemctl restart collectd.service')
             else:
-                self.node.remoter.run('sudo service collectd start')
+                self.node.remoter.run('sudo service collectd restart')
```
Comment (amoskong) [inline sdcm/collectd.py]: Fixed https://github.com/scylladb/scylla-cluster-tests/issues/668

File: upgrade_test.py [hunk 2]
```diff
@@ -118,6 +118,8 @@ def upgrade_node(self, node):
                 else:
                     node.remoter.run('sudo apt-get update')
                     node.remoter.run('sudo apt-get dist-upgrade {} -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" --force-yes --allow-unauthenticated'.format(scylla_pkg))
+        if self.db_cluster.params.get('test_sst3', default=None):
```
Comment (bentsi) [inline upgrade_test.py]: 1. why not `self.params`?
2. when introducing new params please add them to scylla.yaml with explanation what is their purpose

File: sdcm/cluster.py [hunk 3]
```diff
@@ -1845,6 +1850,24 @@ def node_setup(self, node, verbose=False, db_node_address=None, **kwargs):
             self.kill_stress_thread()
             return
 
+        if node.is_ubuntu14():
```
Comment (bentsi) [inline sdcm/cluster.py]: More cleaner and efficient way:

```
install_java_script = dedent("""
    apt-get install software-properties-common -y
    add-apt-repository -y ppa:openjdk-r/ppa
    add-apt-repository -y ppa:scylladb/ppa
    apt-get update
    apt-get install -y openjdk-8-jre-headless
    update-java-alternatives -s java-1.8.0-openjdk-amd64
""")
node.remoter.run('sudo bash -cxe "%s"' % install_java_script)
```
`-x`: will print each command that is executed
`-e`: will exit on first failure

File: upgrade_test.py (hunk 2 above)
Comment (amoskong) [reply upgrade_test.py]: Ok

File: sdcm/cluster.py (hunk 3 above)
Comment (amoskong) [reply sdcm/cluster.py]: Fixed.

**Comment by amoskong** [comment]
Backported this PR to branch-3.0 for making rolling upgrade job PASS.

**Pull request merged**